src/cloudant/replicator.py

```diff
@@ -0,0 +1,171 @@
+
+        @param Database source_db: Database object to replicate from
+        @param Database target_db: Database object to replicate to
+        @param str repl_id: replication_id (I'll create one if you
+            don't specify.)
+
+        Optional overrides (I'll compose these for you, unless you
+            explicitly specify them):
+        @param str/dict source: string or dict representing the source
+            database, along with authentication info, if any.
+        @param str/dict target: string or dict representing the
+            target database, possibly including authentication info.
+        @param dict user_ctx: User to act as.
+
+        Additional params you might specify (I won't pass these along
+            unless specified):
+        @param boolean: create_target: specifies whether or not to
+            create the target, if it doesn't already exist.
+        @param boolean continuous: set to True for a continuous replication.
+
+        """
+
+        data = dict(
+            _id=repl_id if repl_id else unicode(uuid.uuid4()),
+            **kwargs
+        )
+
+        if not data.get('source'):
+            if source_db is None:
+                raise CloudantException(
+                    u"You must specify either a source_db Database "
+                    u"object or a manually composed 'source' string/dict."
+                )
+            data['source'] = {
+                "url": source_db.database_url,
+                "headers": {
+                    "Authorization": source_db.creds['basic_auth']
+                }
+            }
+
+        if not data.get('target'):
+            if target_db is None:
+                raise CloudantException(
+                    u"You must specify either a target_db Database "
+                    u"object or a manually composed 'target' string/dict."
+                )
+            data['target'] = {
+                "url": target_db.database_url,
+                "headers": {
+                    "Authorization": target_db.creds['basic_auth']
+                }
+            }
+
+        if not data.get('user_ctx'):
+            data['user_ctx'] = self.creds['user_ctx']
+
+        return self.create_document(data, throw_on_exists=True)
+
+    def list_replications(self):
+        """
+        _list_replications_
+
+        Returns a list of all replications.
+
+        """
+        docs = self.all_docs(include_docs="true")['rows']
+        return [doc['doc'] for doc in docs]
+
+    def replication_state(self, repl_id):
+        """
+        _replication_state_
+
+        Get the state of the current replication.
+
+        @param str replication_id: id of the replication to inspect.
+
+        """
+        try:
+            repl_doc = self[repl_id]
+        except KeyError:
+            raise CloudantException(
+                "Replication {} not found".format(replication_id))
+        repl_doc.fetch()
+        print "repl_doc: {}".format(repl_doc)
```

Comment:
Whoops. Meant to remove that (artifact of being very puzzled as to why stuff was failing.)